### ISSUE TITLE
Addition of Resonator and Resistive Wall functions as direct passmethods

### ISSUE THREAD

**Issue by lcarver** [atcollab/at#674]
While running tests for #658 , I realised that actually the TRW example is rather slow. One of the reasons for this is likely due to the fact that the function is in python (not directly in C) and the srange has to be specified as an input. When it is like this, on each call to the function you have to first find the appropriate index (binarySearch) and then make the interpolation (getTableWake). For a finely sampled wake or a long wake, this can be a time consuming process.

This is avoided for the BeamLoading module where the longitudinal resonator is directly written in the C (it was done like this to avoid the interpolation around 0 where there is a discontinuity). The Beamloading is therefore fast while the others are slower.

Should we consider adding new PassMethods for each function? This would mean that we make the following additional PassMethods:
LongResonatorPass
TransResonatorPass
TransResWallPass

which may further clutter the atintegrators directory.

**Comment by swhite2401** [comment]
You are correct interpolating large tables is slower than the analytical calculation... this would nevertheless require significant changes in the architecture (both python and C) that is fully based on generating tables.
We should stick to a single element but the analytical case now has to be handled differently, i.e. without table... this will become tricky when adding wake components become you have to pass individual wake parameters to the C.

**Comment by lcarver** [comment]
Do you think it makes sense to have both? We keep the python implementation that we have now, which allows for example a resonator to be added to other wakes with the wake table, and then in addition to this we have a direct C implementation which is good for speed and more accurate (no interpolation).

**Comment by swhite2401** [comment]
I would like to keep thinks general, maintenance is much easier this way

**Comment by lcarver** [comment]
I will think about how to implement this cleanly.
